katdal/test/test_chunkstore_s3.py

```diff
@@ -41,30 +55,36 @@ def start_fakes3(cls, host):
             cls.fakes3 = subprocess.Popen(['fakes3', 'server',
                                            '-r', cls.tempdir, '-p', '0',
                                            '-a', host, '-H', host],
-                                          stdout=subprocess.PIPE,
                                           stderr=subprocess.PIPE)
         except OSError:
             raise SkipTest('Could not start fakes3 server (is it installed?)')
-        start = time.time()
-        # Give up after waiting a few seconds for Fake S3
-        while time.time() - start <= 5:
-            # Look for assigned port number in fakes3 stderr output
-            line = cls.fakes3.stderr.readline().strip()
-            ports_found = re.search(r' port=(\d{4,5})$', line)
-            if ports_found:
-                port = ports_found.group(1)
-                return 'http://%s:%s' % (host, port)
-        raise SkipTest('Could not connect to fakes3 server')
+        # The assigned port number is scraped from stderr and returned via queue
+        port_queue = Queue.Queue()
+        # Ensure that the stderr of fakes3 process is continuously consumed.
+        # This pattern is inspired by the "Launch, Interact, Get Output in
+        # Real Time, Terminate" section of
+        # https://dzone.com/articles/interacting-with-a-long-running-child-process-in-p
+        cls.stderr_consumer = threading.Thread(target=consume_stderr_find_port,
+                                               args=(cls.fakes3, port_queue))
+        cls.stderr_consumer.start()
+        # Give up after waiting a few seconds for Fake S3 to announce its port
+        try:
+            port = port_queue.get(timeout=5)
+        except Queue.Empty:
+            raise SkipTest('Could not connect to fakes3 server')
```

Comment:
Maybe this should be a test failure? If fakes3 started up but we couldn't find a port number, there is probably a bug in the test or in the test environment and we should debug it, rather than thinking everything is okay because Jenkins passed.

Reply:
Right, agreed.

Reply:
Made this an `OSError` instead.